geps/gep-995.md

```diff
@@ -0,0 +1,107 @@
+
+```yaml
+// +kubebuilder:validation:Pattern=`^[a-z][-_a-zA-Z0-9]*$`
+// +kubebuilder:validation:MinLength=1
+// +kubebuilder:validation:MaxLength=253
+```
+
+This format for the `name` field of route rules differs from the pattern adopted for the [`SectionName`](https://github.com/kubernetes-sigs/gateway-api/blob/f544a46ef92b7f234ee3e7bf50da35b05f862c35/apis/v1/shared_types.go#L624C10-L624C10) type, which was thought for specifying mainly DNS subdomain names ([RFC 1123](https://www.rfc-editor.org/rfc/rfc1123)), due to its use in the gateway listeners originally.
+
+### Volition
+
+To preserve backward compatibility with previous version of the affected APIs, the `name` field for route rules should be introduced in the API as optional – i.e., end-user are not forced to add it to their existing or new route objects.
+
+Implementations MAY recomend the usage of the `name` field for enabling specific features, such as for supporting policy attachment targetting individual route rules, and more assertive log messages and/or status reporting that include on the name of the rule. However, because as by API design the presence of the field is optional, implementations MUST take into account that a value may sometimes not be available. For such cases, implementations are free to decide whether to provide the feature depending the `name` field, if the feature is not `core` for compliance, or to enable the feature relying on another method of referencing of choice.
+
+### Default value
+
+Implementations SHOULD not assume a default vaule for the `name` field for route rules when not specified by the end-user.
+
+Please refer to the [Volition](#volition) subsection for alternatives if `name` field is missing.
+
+### Mutability
+
+Implementations MAY prevent end-users from updating the name of a route rule. If updates are allowed the semantics and behaviour will depend on the underlying implementation.
+
+## Conformance Details
+
+(This section describes the names to be used for the feature or
+features in conformance tests and profiles.
+
+These should be `CamelCase` names that specify the feature as
+precisely as possible, and are particularly important for
+Extended features, since they may be surfaced to users.)
+
+## Alternatives
+
+### Reference route rules by index
+- Consistency can be easily broken by adding/removing new rules
+- Less descriptive compared to a name
+
+### Reference route rules by matching field values of rule
+- Consistency can be easily broken due to duplication of values between reference and referent
+- Does not solve for easy referencing in debugging, logging, status messages – i.e. some verbosity required to communicate what route rule the events refers to
+
+### Break down route objects into smaller ones (with less rules)
+- Could lead to an explosion of route objects containing no more than one rule each, thus defeating the purpose of supporting lists of rules within route objects in the first place
```

Comment:
Probably worth citing https://github.com/kubernetes-sigs/gateway-api/issues/1696#issuecomment-1679804122 here too - even though large routes with many rules should be discouraged, this is still useful with small routes with just a few rules.